linkatos/confirmation.py

```diff
@@ -9,12 +9,19 @@ def update_if_url(parsed_message, expecting_confirmation):
     return (expecting_confirmation)
 
 
-def process_if_yn(parsed_message, expecting_confirmation):
-    if expecting_confirmation is False or parsed_message['type'] is not 'yn_answer':
-        is_yes = False
-        return (expecting_confirmation, is_yes)
+def evaluate(expecting_confirmation, parsed_message, item_ts):
+    # returns (@expecting_confirmation: boolean, @confirmation: boolean)
+    if expecting_confirmation is False or \
+       parsed_message['type'] is not 'reaction' or \
+       parsed_message['item_ts'] != item_ts:
+        return (expecting_confirmation, False)
 
-    is_yes = parsed_message['out']  # True if 'yes', False if 'no'
-    expecting_confirmation = False
+    if parsed_message['message'] == '+1':
+        return (False, True)
 
-    return (expecting_confirmation, is_yes)
+    if parsed_message['message'] == '-1':
+        return (False, False)
+
+    # when we're expecting a confirmation and it's not a thumbs up or a thumbs
+    # down, we keep waiting
+    return (expecting_confirmation, False)
```

Comment:
This is a not-so-good attempt to untangle the `expecting_confirmation` logic from the reaction itself.

```python
def evaluate(expecting_confirmation, parsed_message, item_ts):    
    if not expecting_confirmation or parsed_message['type'] != 'reaction':
        return (expecting_confirmation, False)

    reaction = parsed_message['message']
    message_id = parsed_message['item_ts']

    if not known_reaction(reaction) or message_id != item_ts:
        return (expecting_confirmation, False)

     return (False, evaluate_reaction(reaction))


def evaluate_reaction(reaction):
    return reaction == '+1'

def known_reaction(reaction):
    return reaction in ['+1', '-1']
```